src/web/services/email-generation.service.ts

```diff
@@ -0,0 +1,31 @@
+import { Injectable } from '@angular/core';
+import { Observable } from 'rxjs';
+import { ResourceEndpoints } from '../types/api-endpoints';
+import { Email } from '../types/api-output';
+import { HttpRequestService } from './http-request.service';
+
+/**
+ * Handles email generation.
+ */
+@Injectable({
+  providedIn: 'root',
+})
+export class EmailGenerationService {
+
+  constructor(private httpRequestService: HttpRequestService) { }
+
+  /**
+   * Get email contents by calling API.
+   */
+  getEmail(queryParams: {
+    courseId: string, studentemail: string, emailtype: string, fsname?: string,
```

Comment:
Instead of `queryParams` for the method parameter, why not just put all the individual `courseId`, `studentEmail`, etc. as the method parameters?